jdaviz/configs/default/plugins/export_plot/export_plot.py

```diff
@@ -74,3 +104,138 @@ def vue_save_figure(self, filetype):
+            # mitigating sleep wasn't long enough
+            while viewer.figure._upload_png_callback is not None:
+                time.sleep(ts)
+
+        # Grab frame size.
+        frame_shape = cv2.imread(temp_png_files[0]).shape
+        frame_size = (frame_shape[1], frame_shape[0])
+
+        try:
+            video = cv2.VideoWriter(filename, cv2.VideoWriter_fourcc(*'mp4v'), fps, frame_size, True)  # noqa: E501
+            for cur_pngfile in temp_png_files:
+                video.write(cv2.imread(cur_pngfile))
+        finally:
+            cv2.destroyAllWindows()
+            video.release()
+            slice_plg._on_slider_updated({'new': orig_slice})
+            self.app.loading = False
+
+        if rm_temp_files:
+            for cur_pngfile in temp_png_files:
+                os.remove(cur_pngfile)
+
+    def save_movie(self, i_start, i_end, filename=None, filetype=None, rm_temp_files=True):
+        """Save selected slices as a movie.
+
+        This method creates a PNG file per frame (``._cubeviz_movie_frame_<n>.png``)
+        in the working directory before stitching all the frames into a movie.
+        Please make sure you have sufficient memory for this operation.
+        PNG files are deleted after the movie is created unless otherwise specified.
+        If another PNG file with the same name already exists, it will be silently replaced.
+
+        Movie is written out with frame rate of 5 frames per second (fps).
+
+        Parameters
+        ----------
+        i_start, i_end : int
+            Slices to record; each slice will be a frame in the movie.
+            Unlike Python indexing, ``i_end`` is inclusive.
+            Wrapping and reverse indexing are not supported.
+
+        filename : str or `None`
+            Filename for the movie to be recorded. Include path if necessary.
+            If not given, it will be named ``mymovie.mp4`` in the working directory.
+            If another file with the same name already exists, it will be silently replaced.
+
+        filetype : {'mp4', `None`}
+            Currently only MPEG-4 is supported. This keyword is reserved for future support
+            of other format(s).
+
+        rm_temp_files : bool
+            Remove temporary PNG files after movie creation. Default is `True`.
+
+        """
+        if self.config != "cubeviz":
+            raise NotImplementedError(f"save_movie is not available for config={self.config}")
+
+        if not HAS_OPENCV:
+            raise ImportError("Please install opencv-python to save cube as movie.")
+
+        if filetype is None:
+            if filename is not None and '.' in filename:
+                filetype = filename.split('.')[-1]
+            else:
+                # default to MPEG-4
+                filetype = "mp4"
+
+        if filetype != "mp4":
+            raise NotImplementedError(f"filetype={filetype} not supported")

```

Comment:
To also support GIF, yet another dependency (`imageio`) needs to be introduced. I'd rather not have to tackle that in this PR. I think it would be cleaner as a follow-up PR if we really want that.